lib/service/shopware_service.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:dio/dio.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:product_import_app/notifier/access_data_provider.dart';
+import 'package:provider/provider.dart';
+
+class ShopwareService {
+  static ShopwareService _instance;
+  final Dio dio;
+
+  ShopwareService._internal() : dio = Dio() {
+    // todo add interceptor for refreshing token if invalid
+    // todo set base options accept / authorization
+  }
+
+  factory ShopwareService() {
+    if (_instance == null) {
+      _instance = ShopwareService._internal();
+    }
+
+    return _instance;
+  }
+
+  Future uploadProduct(
+    BuildContext context, {
+    @required String name,
+    @required String number,
+    @required num price,
+    String description,
+    @required num taxRate,
+    @required int stock,
+  }) async {
+    final accessData = Provider.of<AccessDataChangeNotifier>(
+      context,
+      listen: false,
+    );
+
+    accessData.init(
```

Comment:
warum hier der hard coded access token? 
kannst du dir den nicht über den getter holen

Reply:
der laeuft alle 10 minuten ab :D 
und man bekommt noch keinen neuen und weil dann ein accessToken in den prefs steht kommt man nicht wieder zum login 
Damit ich das testen konnte hab ich das da erstmal so eingebaut